gravitee-apim-console-webui/src/management/api/proxy/properties-ng/api-properties.component.ts

```diff
@@ -118,19 +107,33 @@ export class ApiPropertiesComponent implements OnInit, OnDestroy {
       .pipe(
         switchMap((api) => {
           if (api.definitionVersion !== 'V1') {
-            const updateApi: UpdateApiV2 | UpdateApiV4 = { ...api, ...this.api };
+            const updateApi: UpdateApiV2 | UpdateApiV4 = {
+              ...api,
+              properties: this.properties,
+              services: {
+                ...this.api.services,
+                dynamicProperty: this.provider,
+              },
+            };
             return this.apiService.update(this.ajsStateParams.apiId, updateApi);
           } else {
             return EMPTY;
           }
         }),
         tap((api: ApiV2 | ApiV4) => {
           this.ajsRootScope.$broadcast('apiChangeSuccess', { api });
-          this.api = api;
+          this.updateApi(api);
           this.isDirty = false;
         }),
         takeUntil(this.unsubscribe$),
       )
       .subscribe();
   }
+
+  private updateApi(api: ApiV2 | ApiV4) {
```

Comment:
Ha the wording confused me I thought it was a real api update 🙈

Reply:
no real better idea. maybe `setApi` or `initCurrentApi`

Reply:
I changed it to `initApiData`